rewrite-maven/src/test/java/org/openrewrite/maven/MavenParserTest.java

```diff
@@ -3982,22 +3982,363 @@ void jaxbRuntime() {
                 </dependencies>
               </project>
               """,
-                spec -> spec.afterRecipe(pom -> {
-                            MavenResolutionResult mrr = pom.getMarkers().findFirst(MavenResolutionResult.class).orElseThrow();
-                            assertThat(mrr.getDependencies().get(Scope.Runtime))
-                                    .map(ResolvedDependency::getGav)
-                                    .map(ResolvedGroupArtifactVersion::asGroupArtifactVersion)
-                                    .as("At one point this test failed with no version number found for jakarta.xml.bind-api because ResolvedPom was not considering classifiers as significant for dependency management")
-                                    .containsExactlyInAnyOrder(
-                                            new GroupArtifactVersion("org.glassfish.jaxb", "jaxb-runtime", "2.3.9"),
-                                            new GroupArtifactVersion("jakarta.xml.bind", "jakarta.xml.bind-api", "2.3.3"),
-                                            new GroupArtifactVersion("org.glassfish.jaxb", "txw2", "2.3.9"),
-                                            new GroupArtifactVersion("com.sun.istack", "istack-commons-runtime", "3.0.12"),
-                                            new GroupArtifactVersion("com.sun.activation", "jakarta.activation", "1.2.2")
-                                    );
-                        }
-                )
+            spec -> spec.afterRecipe(pom -> {
+                  MavenResolutionResult mrr = pom.getMarkers().findFirst(MavenResolutionResult.class).orElseThrow();
+                  assertThat(mrr.getDependencies().get(Scope.Runtime))
+                    .map(ResolvedDependency::getGav)
+                    .map(ResolvedGroupArtifactVersion::asGroupArtifactVersion)
+                    .as("At one point this test failed with no version number found for jakarta.xml.bind-api because ResolvedPom was not considering classifiers as significant for dependency management")
+                    .containsExactlyInAnyOrder(
+                      new GroupArtifactVersion("org.glassfish.jaxb", "jaxb-runtime", "2.3.9"),
+                      new GroupArtifactVersion("jakarta.xml.bind", "jakarta.xml.bind-api", "2.3.3"),
+                      new GroupArtifactVersion("org.glassfish.jaxb", "txw2", "2.3.9"),
+                      new GroupArtifactVersion("com.sun.istack", "istack-commons-runtime", "3.0.12"),
+                      new GroupArtifactVersion("com.sun.activation", "jakarta.activation", "1.2.2")
+                    );
+              }
+            )
           )
         );
     }
+
+    @Nested
+    class DependencyManagement {
```

Comment:
As a quick note: these tests also pass on `main`, so we're not yet reproducing or fixing any issue it seems.